code/campaignclassic/src/main/java/com/adobe/marketing/mobile/AEPPushNotificationBuilder.java

```diff
@@ -0,0 +1,502 @@
+                    notification = createBasicTemplatePushNotification(basicPushTemplate, context);
+                    break;
+//            case AUTO_CAROUSEL:
+//                final CarouselPushTemplate carouselPushTemplate = new CarouselPushTemplate(messageData);
+//                notification = createAutoCarouselTemplatePushNotification(payload, context);
+//                break;
+//            case MANUAL_CAROUSEL:
+//                final CarouselPushTemplate carouselPushTemplate = new CarouselPushTemplate(messageData);
+//                notification = createManualCarouselTemplatePushNotification(payload, context);
+//                break;
+//            case INPUT_BOX:
+//                final InputBoxPushTemplate inputBoxPushTemplate = new InputBoxPushTemplate(messageData);
+//                notification = createInputBoxTemplatePushNotification(payload, context);
+//                break;
+                case UNKNOWN:
+                case LEGACY:
+                default:
+                    final AEPPushTemplate aepPushTemplate = new AEPPushTemplate(messageData);
+                    notification = createLegacyPushNotification(aepPushTemplate, context);
+                    break;
+            }
+        } catch (final IllegalArgumentException exception) {
+            Log.error(LOG_TAG, SELF_TAG, "Failed to create a Notification from the provided AEPPushPayload, an illegal argument exception occurred: %s", exception.getLocalizedMessage());
+            return null;
+        }
+
+        return notification;
+    }
+
+    @NonNull
+    static Notification createLegacyPushNotification(final AEPPushTemplate pushTemplate, final Context context) {
+        final String channelId = createChannelAndGetChannelID(pushTemplate, context);
+        final NotificationCompat.Builder builder = new NotificationCompat.Builder(context, channelId)
+                .setContentTitle(pushTemplate.getTitle())
+                .setContentText(pushTemplate.getBody())
+                .setNumber(pushTemplate.getBadgeCount())
+                .setPriority(pushTemplate.getNotificationPriority())
+                .setAutoCancel(true);
+
+        setLargeIcon(builder, pushTemplate);
+        setSmallIcon(builder, pushTemplate, context); // Small Icon must be present, otherwise the notification will not be displayed.
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            setVisibility(builder, pushTemplate);
+        }
+        addActionButtons(builder, pushTemplate, context); // Add action buttons if any
+        setSound(builder, pushTemplate, context);
+        setNotificationClickAction(builder, pushTemplate, context);
+        setNotificationDeleteAction(builder, pushTemplate, context);
+
+        return builder.build();
+    }
+
+    @NonNull
+    static Notification createBasicTemplatePushNotification(final BasicPushTemplate pushTemplate, final Context context) {
```

Comment:
Because this pattern of `create*TemplatePushNotification` is needed for each template type, and each of these templates have a decent amount of error and view handling, I feel like this class will quickly bulk up with these methods. 
To be easier on the eyes, we can perhaps have one component for each template that is responsible for converting it into a notification. Thoughts? If you agree we can do this as a follow up pr.

Reply:
i agree, the class has gotten quite large and this would alleviate that. i'll get it added in the next pr.